packages/syft/src/syft/core/node/common/node_service/testing_services/smpc_executor_service.py

```diff
@@ -12,30 +13,49 @@
 from ......logger import traceback_and_raise
 from .....store.storeable_object import StorableObject
 from ....abstract.node import AbstractNode
+from ...action.smpc_action_functions import _MAP_ACTION_TO_FUNCTION
 from ...action.smpc_action_message import SMPCActionMessage
-from ...action.smpc_action_message import _MAP_ACTION_TO_FUNCTION
+from ...action.smpc_action_seq_batch_message import SMPCActionSeqBatchMessage
 from ..node_service import ImmediateNodeServiceWithoutReply
 
+SMPCMessageType = Union[Type[SMPCActionMessage], Type[SMPCActionSeqBatchMessage]]
+
 
 class SMPCExecutorService(ImmediateNodeServiceWithoutReply):
     @staticmethod
-    def message_handler_types() -> List[Type[SMPCActionMessage]]:
-        return [SMPCActionMessage]
+    def message_handler_types() -> List[SMPCMessageType]:
+        return [SMPCActionMessage, SMPCActionSeqBatchMessage]
 
     @staticmethod
     def process(
         node: AbstractNode,
-        msg: SMPCActionMessage,
+        msg: Union[SMPCActionMessage, SMPCActionSeqBatchMessage],
         verify_key: Optional[VerifyKey] = None,
     ) -> None:
-        """Given an SMPCAction, execute it (this action is sent to the node
-        by the RabitMQ task)
+        """Given an SMPCAction or SMPCActionSeqBatched, execute it (this
+        action/batch of actions is sent to the node by the RabitMQ task)
 
         Attributes:
             node (AbstractNode): the node that received the message
-            msg (SMPCActionMessage): the message that should be executed
+            msg (SMPCActionMessage or SMPCActionSeqBatchMessage): the message/batch of messages that should be executed
             verify_key (VerifyKey): the verify_key
         """
+        if isinstance(msg, SMPCActionSeqBatchMessage):
+            while msg.smpc_actions:
+                action = msg.smpc_actions[0]
+                SMPCExecutorService.execute_action(node, action, verify_key)
```

Comment:
Q: George, I forgot could you reiterate why we are clubbing `spdz_mask`, `spdz_multiply` , In the current DB logic, when action fails, we store it in the  DB part,we retry the action when a new object arrives,
George, I think when when spdz multiply fails,this keeps the whole SMPCBatchMessage in DB, when a new message arrives,, `spdz_mask` would be executed again, there would be another roundtrip of communication by spdz mask .

Reply:
This was my  first doubt george, 
`When an item arrives we try again execute spdz_mask and if success spdz_mul - if spdz_mul fails then we will store only the spdz_mul in the DB`
By the DB logic, if the spdz mask succeeds and spdz mul fails, since this came from a single `BatchSeqAction` , this would keep spdz mask and spdz multiply both in the DB , and spdz mask would be executed again when a new object arrives.

Reply:
Great thought , george single message make great sense, as  currently have the RunClassMethodSMPCAction is put in queue, which actually creates SMPCActionMessages and put it at back of queue, this create a lot of hassle, which slows our runtime.I think by this we could execute spdz mask and spdz directly in one action directly in RunClassMethodSMPC itself.

Reply:
I think also by this we would not have to couple the SMPC logic in the backend stream which we discussed, as we execute directly

Reply:
This should perfectly fit the greelet logic I am working on.

Reply:
Discussed in slack